code/API_definitions/quality-on-demand.yaml

```diff
@@ -97,12 +105,16 @@ paths:
 
         A `QOS_STATUS_CHANGED` event notification with `qosStatus` as `UNAVAILABLE` will also be send if the network terminates the session before the requested duration expired
 
-        NOTE: in case of a `QOS_STATUS_CHANGED` event with `qosStatus` as `UNAVAILABLE` and `statusInfo` as `NETWORK_TERMINATED` the resources of the QoS session are not directly released, but will get deleted automatically at earliest 360 seconds after the event.
-        This behavior allows clients which are not receiving notification events but are polling to get the session information with
-        the `qosStatus` `UNAVAILABLE` and `statusInfo` `NETWORK_TERMINATED`. Before a client can attempt to create a new QoD session
-        for the same device and flow period they must release the session resources with an explicit `delete` operation if not yet automatically deleted.
+        **NOTES:**
+        - In case of a `QOS_STATUS_CHANGED` event with `qosStatus` as `UNAVAILABLE` and `statusInfo` as `NETWORK_TERMINATED` the resources of the QoS session are not directly released, but will get deleted automatically at earliest 360 seconds after the event.
+
+          This behavior allows clients which are not receiving notification events but are polling to get the session information with the `qosStatus` `UNAVAILABLE` and `statusInfo` `NETWORK_TERMINATED`. Before a client can attempt to create a new QoD session for the same device and flow period they must release the session resources with an explicit `delete` operation if not yet automatically deleted.
+        - The access token may be either 2-legged or 3-legged. If a 3-legged access token is used, the end user identified by the `device` parameter must also be associated with the access token.
```

Comment:
Can a 2-legged token be used at this endpoint at all? Because line 62 also states `It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory.`

Reply:
@sfnuser you have always consider the complete text, including: "Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation." Taking the following sentence out of this context is not helpful.

There are several questions to be clarified for a concrete onboarding, e.g.:
* is personal user data processed (think about IoT use case, but also different interpretations within legal frameworks if this endpoint is "processing" personal information)
* and: does the legal framework require that users can opt-out from the processing? - that's depends for example on the legal base which is used (think for example that the API is used in a private network where all users have signed a specific contract that their data will be processed).

The complexity of these question does not allow to make a final decision independent of the concrete onboarding of an application, hence the text. Independent from the fact that 3-legged flows are on the safe side and should be the default case.

Reply:
@hdamker Thanks. My question was also because in the last [meeting](https://wiki.camaraproject.org/display/CAM/2024-06-14+Quality+on+Demand+-+Meeting+Minutes) we noted that this PR will clarify

`
Endpoints which have different behaviour with 2-legged vs 3-legged
`
`
Endpoints which don't require 3-legged token (no user resource involved)
` 

Yes, we say that onboarding decides the final authorization flow taking into account all the purpose/legal complexity etc. but we also mention in line 62 that in `these` scenarios 3-legged tokens becomes mandatory :-(. When a specification mentions something is `mandatory`, developers tend to think it like a MUST field interpretation in a RFC. We could have avoided line 62 completely, which is more like a moral statement. These reasoning and examples you have quoted can be captured in some official documentation as valid examples. All the specs can refer to it for the reasoning behind vague disposition of security in the specification.

Again, it is not specific to this PR or QoD alone. I think it is too late considering the release timelines. So I will let it pass :-)

Reply:
@sfnuser line 62 has two conditions which are connected with "and". The second one is _" and users can exercise their rights through mechanisms such as opt-in and/or opt-out"_. In case the users have such rights given by the specific legal framework and for the specific use case (purpose) than it is indeed de-facto mandatory to have the user within the flow. At least that is my interpretation of the statement. If this statement is needed here is another discussion, which belongs into ICM.

Reply:
@sfnuser Resolved this conversation based on your statement that you "will let it pass" for now.